src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -520,7 +520,7 @@ public DatasetSummary retrieveSummaryByName(String name) {
             StringUtils.join(whereClauses, " AND ");
         Integer total = jdbcTemplate.queryForObject(countSql, params, Integer.class);
         if (total == null) {
-            throw new CorruptMetadataException("Impossible null value from count");
+            throw new CorruptMetadataException("Impossible null value from total count");
```

Comment:
I don't really understand why we bother throwing and don't just return a 0 in this case

Reply:
I think its because we expect 0, and we're working around `queryForObject` not being annotated `@NotNull`.